src/Lib.P2P/Peer.cs

```diff
@@ -0,0 +1,186 @@
+
+        /// <summary>
+        ///   The public key of the node.
+        /// </summary>
+        /// <value>
+        ///   The base 64 encoding of the node's public key.  The default is <b>null</b>
+        /// </value>
+        /// <remarks>
+        ///   The IPFS public key is the base-64 encoding of a protobuf encoding containing 
+        ///   a type and the DER encoding of the PKCS Subject Public Key Info.
+        /// </remarks>
+        /// <seealso href="https://tools.ietf.org/html/rfc5280#section-4.1.2.7"/>
+        public string PublicKey { get; set; }
+
+        /// <summary>
+        ///   The multiple addresses of the node.
+        /// </summary>
+        /// <value>
+        ///   Where the peer can be found.  The default is an empty sequence.
+        /// </value>
+        public IEnumerable<MultiAddress> Addresses { get; set; } = _noAddress;
+
+        /// <summary>
+        ///   The name and version of the IPFS software.
+        /// </summary>
+        /// <value>
+        ///   For example "go-ipfs/0.4.17/".
+        /// </value>
+        /// <remarks>
+        ///   There is no specification that describes the agent version string.  The default
+        ///   is "unknown/0.0".
+        /// </remarks>
+        public string AgentVersion { get; set; } = Unknown;
+
+        /// <summary>
+        ///  The name and version of the supported IPFS protocol.
+        /// </summary>
+        /// <value>
+        ///   For example "ipfs/0.1.0".
+        /// </value>
+        /// <remarks>
+        ///   There is no specification that describes the protocol version string. The default
+        ///   is "unknown/0.0".
+        /// </remarks>
+        public string ProtocolVersion { get; set; } = Unknown;
+
+        /// <summary>
+        ///   The <see cref="MultiAddress"/> that the peer is connected on.
+        /// </summary>
+        /// <value>
+        ///   <b>null</b> when the peer is not connected to.
+        /// </value>
+        public MultiAddress ConnectedAddress { get; set; }
+
+        /// <summary>
+        /// The round-trip time it takes to get data from the peer.
+        /// </summary>
+        public TimeSpan? Latency { get; set; }
+
+        /// <summary>
+        ///   Determines if the information on the peer is valid.
+        /// </summary>
+        /// <returns>
+        ///   <b>true</b> if all validation rules pass; otherwise <b>false</b>.
+        /// </returns>
+        /// <remarks>
+        ///    Verifies that
+        ///    <list type="bullet">
+        ///      <item><description>The <see cref="Id"/> is defined</description></item>
+        ///      <item><description>The <see cref="Id"/> is a hash of the <see cref="PublicKey"/></description></item>
+        ///    </list>
+        /// </remarks>
+        public bool IsValid()
+        {
+            if (Id == null)
+                return false;
+            if (PublicKey != null && !Id.Matches(Convert.FromBase64String(PublicKey)))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)